torchquantum/functional/func_controlled_unitary.py

```diff
@@ -97,7 +98,7 @@ def controlled_unitary(
     n_wires = n_c_wires + n_t_wires
 
     # compute the new unitary, then permute
-    unitary = torch.tensor(torch.zeros(2**n_wires, 2**n_wires, dtype=C_DTYPE))
+    unitary = torch.zeros(2**n_wires, 2**n_wires, dtype=C_DTYPE).clone().detach()
```

Comment:
One more comment — why are you cloning + detaching here? Shouldn’t `torch.zeros` create a new tensor?

Reply:
Oh, I forgot to mention about this. The reason I changed this code is to fix the following `UserWarning` from PyTorch:

`To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor)`

Reply:
Oh, wait, I should have just removed `torch.tensor()` here. It is fixed now.